codex/manifest/manifest.nim

```diff
@@ -58,6 +63,42 @@ iterator pairs*(self: Manifest): tuple[key: int, val: Cid] =
 func contains*(self: Manifest, cid: Cid): bool =
   cid in self.blocks
 
+
+############################################################
+# Various sizes and verification
+############################################################
+
+func bytes*(self: Manifest, pad = true): int =
+  ## Compute how many bytes corresponding StoreStream(Manifest, pad) will return
+  if pad or self.protected:
+    self.len * self.blockSize
+  else:
+    self.originalBytes
+
+func rounded*(self: Manifest): int =
+  ## Number of data blocks in *protected* manifest including padding at the end
+  roundUp(self.originalLen, self.K)
+
+func steps*(self: Manifest): int =
+  ## Number of EC groups in *protected* manifest
+  divUp(self.originalLen, self.K)
+
+func verify*(self: Manifest) =
+  ## Check manifest correctness
+  ##
+  let originalLen = (if self.protected: self.originalLen else: self.len)
+
+  if divUp(self.originalBytes, self.blockSize) != originalLen:
+    raise newException(Defect, "Broken manifest: wrong originalBytes")
```

Comment:
We shouldn't raise `Defect` here or anywhere where the data might come from the network, in this case, if a malformed manifest is sent to us it will crash the application. I would use our standard `?!void` here with an appropriate error message.